src/python/pants/backend/python/typecheck/mypy/subsystem.py

```diff
@@ -258,9 +262,12 @@ async def mypy_first_party_plugins(mypy: MyPy) -> MyPyFirstPartyPlugins:
     )
 
     requirements = PexRequirements.create_from_requirement_fields(
-        plugin_tgt[PythonRequirementsField]
-        for plugin_tgt in transitive_targets.closure
-        if plugin_tgt.has_field(PythonRequirementsField)
+        (
+            plugin_tgt[PythonRequirementsField]
+            for plugin_tgt in transitive_targets.closure
+            if plugin_tgt.has_field(PythonRequirementsField)
+        ),
+        constraints_strings=(str(constraint) for constraint in global_requirement_constraints),
```

Comment:
I don't think this is right. Tool lockfiles don't use the constraints file: https://github.com/pantsbuild/pants/pull/12639

Reply:
The default for `apply_constraints` was `True`, so this position _was_ using the lockfile. Since these are first-party (i.e. source) plugins, it seemed reasonable to apply the constraints, but let me know.

Reply:
...Ah, but in all cases these requests are just using `PexRequirements` to collect the requirement strings. Will fix.